### PR TITLE
Add Dialogs in the EditQuizActivtiy and visual separation

### PR DESCRIPTION
**Pull request by nbaehler**
I added three `ConfirmationDialogs`:

1. for confirming the deletion of question from the quiz
2. for confirming exiting the edition mode without saving
3. for confirming to end the edition of the quiz with saved changes.

I had to add a button in the menu for this.

Additionally I added a line that separates the `EditQuizOverview` fragment from the `EditQuizPreview` fragment. As we're using the same layout for the answering quizzes workflow the problem is fixed there as well.

Finally I had planned to use `InfoDialogs` which simply show a message and have a button to close the `Dialog` but I didn't need them after all. I had already implemented them and they are going to useful to replace Toast so I keep the code anyway.

### CODE REVIEW

**Comment by nbaehler** [comment]
Finally I added a `InfoDialog` that tells the user that he can't save an empty quiz.

File: app/src/main/res/values-fr/strings.xml [hunk 1]
```diff
@@ -79,7 +79,14 @@
     <string name="done">Ok</string>
     <string name="yes">Oui</string>
     <string name="no">Non</string>
-    <string name="warning_delete">Êtes-vous sûr de vouloir supprimer ce quiz ? Cela va supprimer toutes les questions de ce quiz.</string>
+    <string name="add_quiz_message">Entrez le nom de votre quiz</string>
+    <string name="edit_title">Renommer</string>
+    <string name="warning_delete_quiz">Êtes-vous sûr de vouloir supprimer ce quiz ? Cela va supprimer toutes les questions de ce quiz.</string>
+    <string name="warning_delete_question">Êtes-vous sûr de vouloir supprimer cette question ?</string>
+    <string name="warning_exit_edition">Êtes-vous sûr de vouloir annuler l\'édition du quiz ? Les changements que vous avez effectué vont être ignorés.</string>
```
Comment (Anthony-Iozzia) [inline app/src/main/res/values-fr/strings.xml:86]: Typo: "que vous avez effectués"

File: app/src/main/java/ch/epfl/qedit/view/home/HomeQuizListFragment.java [hunk 2]
```diff
@@ -296,6 +296,9 @@ public void onActivityResult(int requestCode, int resultCode, Intent data) {
                         (StringPool) data.getExtras().getSerializable(STRING_POOL);
                 // TODO send back to data base etc.
             }
+            // When the user decides to stop the edition without saving the changes the
+            // EditQuizActivity will return with RESULT_CANCELED. But we do not need to do anything
+            // in thing in this case.
```
[the remark this answers is not in the document]
Comment (nbaehler) [reply app/src/main/java/ch/epfl/qedit/view/home/HomeQuizListFragment.java:301]: I corrected the typo.

File: app/src/main/res/values-fr/strings.xml [hunk 3]
```diff
@@ -54,7 +54,7 @@
     <string name="menu_timer">Minuteur</string>
     <string name="menu_previous">Précédent</string>
     <string name="menu_next">Suivant</string>
-    <string name="menu_done">Terminer</string>
+    <string name="menu_done">Terminer et sauvgarder</string>
```
[the remark this answers is not in the document]
Comment (nbaehler) [reply app/src/main/res/values-fr/strings.xml:57]: I corrected the typo.

File: app/src/main/res/values-fr/strings.xml (hunk 1 above)
Comment (nbaehler) [reply app/src/main/res/values-fr/strings.xml:86]: I corrected the typo.

**Comment by nbaehler** [comment]
The map test is still failing on the CI, locally all is OK. When I comment the test the check passes. But I'm bringing down the diff-coverage in doing so, hence I can't merge anyway.

**Comment by brunnerant** [comment]
If it is really impossible to make that test pass, we might as well merge it as is. I don't think it would lower the coverage that much if only one test needs to be commented.

**Comment by nbaehler** [comment]
> If it is really impossible to make that test pass, we might as well merge it as is. I don't think it would lower the coverage that much if only one test needs to be commented.

I've finally commented the line that causes the test to fail. I do not know why it works locally but not on the CI. During debugging I made some changes to the `EditMapsActivity` by adding a container for the map `Fragment` and then perform a transaction to put it inside of the container. Currently I'm waiting on the OK from @cjordan7 as he's the one working on the map.

**Comment by nbaehler** [comment]
The same type of test, checking that the activities label is shown correctly using this line:
`onView(withText(R.string.title_activity_edit_quiz)).check(matches(isDisplayed()));`
fails on the CI (and on the CI only) for the `EditQuizActivity`. I have no idea why @cjordan7 could merge this test in the first place. I'm going to branch of from master and test the same thing on a clean branch.

**Comment by nbaehler** [comment]
As I learned that the class that I had been changing while debugging was generated by Android Studio I'm going to revert to the last commit passing the checks.

**Comment by nbaehler** [comment]
> The same type of test, checking that the activities label is shown correctly using this line:
> `onView(withText(R.string.title_activity_edit_quiz)).check(matches(isDisplayed()));`
> fails on the CI (and on the CI only) for the `EditQuizActivity`. I have no idea why @cjordan7 could merge this test in the first place. I'm going to branch off from master and test the same thing on a clean branch.

On the clean branch the check actually passed.

**Comment by nbaehler** [comment]
Reverting seems to be a bad choice as it commits the "inverses" of the commit that are reverted. I do not want to undo the changes that came from merges with master on master itself. I'm going to reset instead

**Comment by nbaehler** [comment]
I do not know what's going on, maybe I didn't worship the goddess of GIT with enough conviction ... The checks still run the tests which are no longer on the branch, at least at the current head.

**Comment by nbaehler** [comment]
Well, I give up on this PR and this branch. I will port the code to a new issue, a new branch and a new PR, directly the code not the commits. I don't know what corrupted this branch but I can't afford loosing even more time with this.

**Comment by nbaehler** [comment]
We have finally found the source of the problem that I had on this PR. In the menu on the top right corner of the edit workflow I had added a button. The screen on the CI is much smaller so these buttons where now actually covering up the label which made the test fail. Locally the screen is bigger so that's why I couldn't reproduce the bug. We decided to simply remove the test as we now know why he failed. It adds no coverage and we didn't test the label in any other activity which is simply set in the `AndroidManifest`.

**Pull request merged**